src/mapgen.h

```diff
@@ -8,6 +8,7 @@
 #include <vector>
 
 #include "int_id.h"
+#include "editmap.h"
```

Comment:
I don't get this at all. It seems all you need to change here is a forward declaration of `map_extra_trigger`.